### PR TITLE
Add check-commits-count script

### PR DESCRIPTION
**Pull request by enriquebelarte**
This script checks that each PR for driver-toolkit has a single commit.

### CODE REVIEW

**Comment by ybettan** [comment]
If it passes then lgtm.

File: ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml [hunk 1]
```diff
@@ -44,6 +44,11 @@ tests:
   commands: make verify
   container:
     from: src
+- as: check-commits-count
+   commands: /ci/prow/check-commits-count
+   container:
+     from: docker.io/alpine/git:latest
```
Comment (ybettan) [inline ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml:50]: Seems like this is not working https://docs.ci.openshift.org/docs/how-tos/external-images/.
A simple work around will be to just `dnf install git` before running `ci/prow/check-commit-count`. WDYT?

File: ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml [hunk 2]
```diff
@@ -44,6 +44,11 @@ tests:
   commands: make verify
   container:
     from: src
+- as: check-commits-count
+   commands: /ci/prow/check-commits-count
```
Comment (ybettan) [inline ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml:48]:
```suggestion
   commands: ci/prow/check-commits-count
```

**Comment by ybettan** [comment]
Seems like it doesn't let us install `git`:
```
ovl: Error while doing RPMdb copy-up:
[Errno 13] Permission denied: '/var/lib/rpm/.dbenv.lock'
You need to be root to perform this command.
```

**Comment by enriquebelarte** [comment]
> Seems like it doesn't let us install `git`:
> 
> ```
> ovl: Error while doing RPMdb copy-up:
> [Errno 13] Permission denied: '/var/lib/rpm/.dbenv.lock'
> You need to be root to perform this command.
> ```

Yes, I'm checking on it. Thank you.

File: ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml [hunk 3]
```diff
@@ -44,6 +48,13 @@ tests:
   commands: make verify
   container:
     from: src
+- as: check-commits-count
+  commands: |
+    sudo yum -y install git
```
Comment (droslean) [inline ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml:53]: You need to do this in your image.

File: ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml [hunk 4]
```diff
@@ -12,6 +16,13 @@ images:
   dockerfile_path: Dockerfile
   from: base
   to: driver-toolkit
+- dockerfile_literal: |
```
Comment (droslean) [inline ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml:19]: You could just add this in your Dockerfile for the `root` image.

Comment (droslean) [reply ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml:19]: Ah ignore me. I just saw that you are not building your root. https://github.com/openshift/driver-toolkit/blob/master/.ci-operator.yaml

Comment (enriquebelarte) [reply ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml:19]: Done. Thank you.

File: core-services/image-mirroring/openshift/mapping_origin_4_12 [hunk 5]
```diff
@@ -145,6 +145,7 @@ registry.ci.openshift.org/origin/4.12:gcp-pd-csi-driver quay.io/openshift/origin
 registry.ci.openshift.org/origin/4.12:gcp-pd-csi-driver-operator quay.io/openshift/origin-gcp-pd-csi-driver-operator:4.12 quay.io/openshift/origin-gcp-pd-csi-driver-operator:4.12.0 quay.io/openshift/origin-gcp-pd-csi-driver-operator:latest
 registry.ci.openshift.org/origin/4.12:gcp-pd-csi-driver-operator-test quay.io/openshift/origin-gcp-pd-csi-driver-operator-test:4.12 quay.io/openshift/origin-gcp-pd-csi-driver-operator-test:4.12.0 quay.io/openshift/origin-gcp-pd-csi-driver-operator-test:latest
 registry.ci.openshift.org/origin/4.12:ghostunnel quay.io/openshift/origin-ghostunnel:4.12 quay.io/openshift/origin-ghostunnel:4.12.0 quay.io/openshift/origin-ghostunnel:latest
+registry.ci.openshift.org/origin/4.12:git-client quay.io/openshift/origin-git-client:4.12 quay.io/openshift/origin-git-client:4.12.0 quay.io/openshift/origin-git-client:latest
```
Comment (droslean) [inline core-services/image-mirroring/openshift/mapping_origin_4_12:148]: Please remove that and add that image in your promotion config as `excluded_images:`

File: ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml [hunk 6]
```diff
@@ -3,6 +3,10 @@ base_images:
     name: "4.12"
     namespace: ocp
     tag: base
+  os:
+    name: centos
+    namespace: origin
+    tag: "8"
```
Comment (ybettan) [inline ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml:9]: Why do we need this?

File: ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml [hunk 7]
```diff
@@ -12,6 +16,13 @@ images:
   dockerfile_path: Dockerfile
   from: base
   to: driver-toolkit
+- dockerfile_literal: |
+    FROM src
+    RUN PACKAGES="git" && \
+        yum install --setopt=tsflags=nodocs -y $PACKAGES && \
```
Comment (ybettan) [inline ci-operator/config/openshift/driver-toolkit/openshift-driver-toolkit-master.yaml:22]: Can we just run `yum install ... -y git &&` instead?

File: core-services/image-mirroring/openshift/mapping_origin_4_12 (hunk 5 above)
Comment (enriquebelarte) [reply core-services/image-mirroring/openshift/mapping_origin_4_12]: Done.Thanks for your for help.

**Comment by droslean** [comment]
/lgtm
/approve

You can rehearse the job after https://github.com/openshift/driver-toolkit/pull/104 is merged.
/hold

**Comment by ybettan** [comment]
LGTM to :)

**Pull request merged**